<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav)"/>

  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch"/>

  <arg name="simulation" default="$(optenv SIMULATION false)" />
  <arg unless="$(arg simulation)" name="image_topic" value="/$(arg uav_name)/rs_d435/color/image_rect_color"/>
  <arg     if="$(arg simulation)" name="image_topic" value="/$(arg uav_name)/realsense/camera/color/image_raw"/>
  <arg unless="$(arg simulation)" name="cinfo_topic" value="/$(arg uav_name)/rs_d435/color/camera_info"/>
  <arg     if="$(arg simulation)" name="cinfo_topic" value="/$(arg uav_name)/realsense/camera/color/camera_info"/>
  <arg unless="$(arg simulation)" name="dbgloc_topic" value="/$(arg uav_name)/uav_localization/dbg_hypotheses"/>
  <arg     if="$(arg simulation)" name="dbgloc_topic" value="/$(arg uav_name)/uav_localization/dbg_hypotheses"/>
  <arg unless="$(arg simulation)" name="dets_topic" value="/$(arg uav_name)/uav_detection/detections"/>
  <arg     if="$(arg simulation)" name="dets_topic" value="/$(arg uav_name)/uav_detection/detections"/>

  <group ns="$(arg uav_name)">
    
    <node pkg="uav_localize" name="backproject_location" type="backproject_display_node" output="screen" launch-prefix="$(arg launch_prefix)" >

      <param name="detection_timeout" value="0.2" />
      <remap from="~image_rect" to="$(arg image_topic)" />
      <remap from="~camera_info" to="$(arg cinfo_topic)" />
      <remap from="~dbg_hypotheses" to="$(arg dbgloc_topic)" />
      <remap from="~detections" to="$(arg dets_topic)" />
      <remap from="~ground_truth" to="/uav4/odometry/rtk_local_odom" />
      <remap from="~odometry" to="/uav42/odometry/rtk_local_odom" />

    </node>

  </group>

</launch>
